frontend/src/stores/studentStore.js

```diff
@@ -0,0 +1,139 @@
+import { action, makeObservable, observable, computed } from 'mobx';
+import service from '../service';
+
+class StudentStore {
+  constructor() {
+    makeObservable(this, {
+      topResults: observable,
+      students: observable,
+      bottomResults: observable,
+      loadStudents: action,
+      setTopResults: action,
+      setBottomResults: action,
+      popLoad: action,
+      pushLoad: action,
+      isLoading: computed,
+    });
+  }
+
+  students = null;
+  topResults = [];
+  bottomResults = [];
+
+  popLoad = () => (this.loading -= 1);
+
+  pushLoad = () => (this.loading += 1);
+
+  setTopResults = (topResults) => (this.topResults = topResults);
+
+  setBottomResults = (bottomResults) => (this.bottomResults = bottomResults);
+
+  loadStudents = async (classId) => {
+    this.pushLoad();
+    const response = await service.loadStudents(classId);
+    if (response) {
+      this.students = response.body.data;
+    }
+
+    const studentsResults = this.computeDistribution(this.students);
+    const topResults = this.computeTopBottom(
+      studentsResults[0],
```

Comment:
Ik weet op dit moment als developer niet zo snel waar `studentsResults[0]` en `studentsResults[1]` naar refereren. Om dit wat explicieter te maken, zou je kunnen noteren op regel 38 hier:

`const [naam1, naam2] = this.computeDistribution(this.students);`

`naam1` refereert dan naar `studentsResults[0]` en `naam2` naar `studentsResults[1]`

Reply:
oh chill, ja ik wist dus niet of dit in javascript kon (want ben gewend met python gewoon idd meerdere variables te returnen) maar chill